Maple2.Server.DebugGame/Graphics/DebugFieldRenderer.cs

```diff
@@ -399,4 +475,520 @@ public void DetachWindow(DebugFieldWindow window) {
+            foreach (KeyValuePair<string, int> kvp in entityCounts) {
+                ImGui.Text($"  {kvp.Key}: {kvp.Value}");
+            }
+
+            ImGui.Separator();
+        }
+        ImGui.End();
+    }
+
+    private void SetFieldOverviewCamera() {
+        var allColliders = GetAllEntities().OfType<FieldBoxColliderEntity>().ToList();
+        if (allColliders.Count > 0) {
+            // Calculate field bounds in map coordinates
+            Vector3 min = allColliders[0].Position;
+            Vector3 max = allColliders[0].Position;
+
+            foreach (var collider in allColliders) {
+                min = Vector3.Min(min, collider.Position - collider.Size * 0.5f);
+                max = Vector3.Max(max, collider.Position + collider.Size * 0.5f);
+            }
+
+            // Transform to visualization coordinates
+            Vector3 centerMap = (min + max) * 0.5f;
+            Vector3 sizeMap = max - min;
+
+            Vector3 center = TransformMapPosition(centerMap);
+            Vector3 size = TransformMapSize(sizeMap);
+            float distance = Math.Max(size.X, Math.Max(size.Y, size.Z)) * 1.5f;
+
+            Context.SetCameraTarget(center);
+            Context.SetCameraPosition(center + new Vector3(0, -distance * 0.7f, distance * 0.7f));
+        } else {
+            // Fallback if no colliders found
+            Context.SetCameraPosition(new Vector3(0, -1000, 1000));
+            Context.SetCameraTarget(Vector3.Zero);
+        }
+    }
+
+    private void UpdateCameraFollow() {
+        if (!Context.IsFollowingPlayer || Context.FollowedPlayerId == null) {
+            return;
+        }
+
+        // Find the player being followed
+        FieldPlayer? followedPlayer = Field.Players.Values.FirstOrDefault(p => p.Value.Character.Id == Context.FollowedPlayerId);
+        if (followedPlayer == null) {
+            // Player not found, stop following
+            Context.StopFollowingPlayer();
+            return;
+        }
+
+        // Update camera to follow player position
+        Vector3 playerPosition = followedPlayer.Position;
+        Vector3 offset = new Vector3(-900, -1200, 2000); // Much further behind and above the player for better overview
+
+        Context.SetCameraTarget(playerPosition);
+        Context.SetCameraPosition(playerPosition + offset);
+    }
+
+    private Vector2D<int> GetFieldWindowSize() {
+        // Get the size from the first active field window, or use default
+        activeMutex.WaitOne();
+        try {
+            var firstWindow = activeWindows.FirstOrDefault();
+            if (firstWindow?.DebuggerWindow != null) {
+                return firstWindow.DebuggerWindow.FramebufferSize;
+            }
+        } finally {
+            activeMutex.ReleaseMutex();
+        }
+
+        // Fallback to default field window size
+        return DebugGraphicsContext.DefaultFieldWindowSize;
+    }
+
+    private Vector2? WorldToScreen(Vector3 worldPos) {
```

Comment:
Could probably be refactored to something like `bool TryWorldToScreen(Vector3 worldPos, out Vector2 screenPos)`